.github/workflows/main.yml

```diff
@@ -75,79 +134,159 @@ jobs:
           curl --insecure --retry 6 --fail --silent --location "https://dl.min.io/client/mc/release/linux-$(dpkg --print-architecture)/archive/mc.RELEASE.2023-06-19T19-31-19Z.sha256sum" | sha256sum --check -
           mv mc.RELEASE.2023-06-19T19-31-19Z mc
           chmod +x mc
-
-      # Run tests and builds image
+      - name: Download artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: latest-njs
+          path: /tmp
+      - name: Load image
+        run: |
+          docker load --input /tmp/latest-njs.tar
+          docker tag nginx-s3-gateway:latest-njs-oss nginx-s3-gateway
       - name: Run tests - latest njs version
         run: ./test.sh --latest-njs --type oss
-      # latest-njs-oss image push [Github]
-      - name: Tag container image for Push to github [latest-njs-oss date]
-        run: docker tag nginx-s3-gateway:latest-njs-oss docker.pkg.github.com/$GITHUB_REPOSITORY/nginx-oss-s3-gateway:latest-njs-oss-${{ steps.date.outputs.date }}
-      - name: Tag container image for Push to github [latest-njs-oss]
-        run: docker tag nginx-s3-gateway:latest-njs-oss docker.pkg.github.com/$GITHUB_REPOSITORY/nginx-oss-s3-gateway:latest-njs-oss
-      - name: Push container image to github [latest-njs-oss date]
-        run: docker push docker.pkg.github.com/$GITHUB_REPOSITORY/nginx-oss-s3-gateway:latest-njs-oss-${{ steps.date.outputs.date }}
-      - name: Push container image to github [latest-njs-oss]
-        run: docker push docker.pkg.github.com/$GITHUB_REPOSITORY/nginx-oss-s3-gateway:latest-njs-oss
-
-      # Run tests and builds image
-      - name: Run tests - stable njs version - unprivileged process
+
+  build-unprivileged-for-test:
+    runs-on: ubuntu-latest
+    needs: test-oss
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver: docker
+      - name: Download artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: oss
+          path: /tmp
+      - name: Load image
+        run: |
+          docker load --input /tmp/oss.tar
+      - name: Build and load oss image
+        uses: docker/build-push-action@v5
+        with:
+          file: Dockerfile.unprivileged
+          context: .
+          tags: nginx-s3-gateway:unprivileged-oss
+          load: true
+      # Save manually here since we need to use the `docker` buildx `driver` but that can't output
+      # a file that upload-artifact likes.
+      - name: save image
+        run: |
+          docker save nginx-s3-gateway:unprivileged-oss > /tmp/unprivileged.tar
+      - name: Upload artifact - unprivileged
+        uses: actions/upload-artifact@v3
+        with:
+          name: unprivileged
+          path: /tmp/unprivileged.tar
```

Comment:
Refresh my memory if the uploaded artifacts eventually get cleaned up. I have a vague memory that they do.

Reply:
There's a default setting for the repo which I believe is 30 days.  However based on this comment I'm going to update the retention explicitly since so it's obvious from the script. 

`retention-days: 1` One day is the minimum which is really too long for this but should serve us fine.

Also worth noting that artifacts are [only accessible from the workflow in which they were created](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#downloading-artifacts-during-a-workflow-run) so there's no chance of us pulling an old artifact which I was worried about initially.